### PR TITLE
Support large scale jitter

### PR DESCRIPTION
**Pull request by ZwwWayne**
## Motivation

Support stronger baselines of Mask R-CNN that uses large-scale jitter with long training schedule.

## Modification

Several configs are added.
As the baseline also requires deeper RPN head with two shared convolutional layers, RPNHead is modified without BC-breaking.
Relies on https://github.com/open-mmlab/mmcv/pull/1205

## BC-breaking (Optional)

Handled to be compatible with the original case.

### CODE REVIEW

File: configs/common/lsj_100e_coco_instance.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
+_base_ = '../_base_/default_runtime.py'
+# dataset settings
+dataset_type = 'CocoDataset'
+data_root = 'data/coco/'
+img_norm_cfg = dict(
+    mean=[123.675, 116.28, 103.53], std=[58.395, 57.12, 57.375], to_rgb=True)
+image_size = (1024, 1024)
+# In mstrain 3x config, img_scale=[(1333, 640), (1333, 800)],
```
Comment (hhaAndroid) [inline configs/common/lsj_100e_coco_instance.py:8]: comment not needed

File: configs/common/lsj_100e_coco_instance.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    dict(type='LoadImageFromFile'),
+    dict(type='LoadAnnotations', with_bbox=True, with_mask=True),
+    dict(
+        type='Resize',
+        img_scale=image_size,
+        ratio_range=(0.1, 2.0),
+        multiscale_mode='range',
+        keep_ratio=True),
+    dict(
+        type='RandomCrop',
```
Comment (hhaAndroid) [inline configs/common/lsj_100e_coco_instance.py:20]: Need to check whether the effect is consistent with FixOutputSize

File: configs/common/lsj_50e_coco_instance.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
+_base_ = '../_base_/default_runtime.py'
```
Comment (hhaAndroid) [inline configs/common/lsj_50e_coco_instance.py:1]: Why not directly inherit 100e.

File: configs/strong_baselines/mask_rcnn_r50_fpn_syncbn-all_lsj_100e_coco.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
+_base_ = [
+    '../_base_/models/mask_rcnn_r50_fpn.py',
+    '../common/lsj_100e_coco_instance.py'
+]
+
+norm_cfg = dict(type='SyncBN', requires_grad=True)
+head_norm_cfg = dict(type='NaiveSyncBN', stats_mode='N', requires_grad=True)
+model = dict(
+    pretrained=None,
+    backbone=dict(frozen_stages=-1, norm_eval=False, norm_cfg=norm_cfg),
```
Comment (hhaAndroid) [inline configs/strong_baselines/mask_rcnn_r50_fpn_syncbn-all_lsj_100e_coco.py:10]: backbone=dict(frozen_stages=-1, norm_eval=False, norm_cfg=norm_cfg,init_cfg=None),

File: configs/strong_baselines/mask_rcnn_r50_fpn_syncbn-all_lsj_100e_coco.py [hunk 5]
```diff
@@ -0,0 +1,21 @@
+_base_ = [
+    '../_base_/models/mask_rcnn_r50_fpn.py',
+    '../common/lsj_100e_coco_instance.py'
+]
+
+norm_cfg = dict(type='SyncBN', requires_grad=True)
+head_norm_cfg = dict(type='NaiveSyncBN', stats_mode='N', requires_grad=True)
+model = dict(
+    pretrained=None,
```
Comment (hhaAndroid) [inline configs/strong_baselines/mask_rcnn_r50_fpn_syncbn-all_lsj_100e_coco.py:9]: delete

File: configs/strong_baselines/mask_rcnn_r50_fpn_syncbn-all_lsj_100e_coco.py [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+    backbone=dict(frozen_stages=-1, norm_eval=False, norm_cfg=norm_cfg),
+    neck=dict(norm_cfg=norm_cfg),
```
Comment (hhaAndroid) [inline configs/strong_baselines/mask_rcnn_r50_fpn_syncbn-all_lsj_100e_coco.py:11]: No need for neck because there is no BN.

File: mmdet/models/dense_heads/rpn_head.py [hunk 7]
```diff
@@ -22,14 +23,28 @@ class RPNHead(AnchorHead):
...
+            rpn_convs = []
+            for i in range(self.num_convs):
+                rpn_convs.append(
+                    ConvModule(
+                        self.in_channels,
+                        self.feat_channels,
+                        3,
+                        padding=1,
+                        inplace=False))
+            self.rpn_conv = nn.Sequential(*rpn_convs)
```
Comment (hhaAndroid) [inline mmdet/models/dense_heads/rpn_head.py:44]: Need to pay attention to the relu operation in the forward code.

File: mmdet/models/utils/naive_syncbn.py [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+                ],
+                                dim=0)
+            vec = AllReduce.apply(vec * B)
+
+            total_batch = vec[-1].detach()
+            momentum = total_batch.clamp(
+                max=1) * self.momentum  # no update if total_batch is 0
+            mean, meansqr, _ = torch.split(vec / total_batch.clamp(min=1),
+                                           C)  # avoid div-by-zero
+
```
Comment (hhaAndroid) [inline mmdet/models/utils/naive_syncbn.py:95]: Missing unit tests.

File: configs/strong_baselines/mask_rcnn_r50_fpn_syncbn-all_lsj_100e_coco.py (hunk 6 above)
Comment (ZwwWayne) [reply configs/strong_baselines/mask_rcnn_r50_fpn_syncbn-all_lsj_100e_coco.py:11]: this will make FPN use syncBN, which is necessary

File: mmdet/models/dense_heads/rpn_head.py (hunk 7 above)
Comment (ZwwWayne) [reply mmdet/models/dense_heads/rpn_head.py:44]: relu is handled in ConvModule

**Review by hhaAndroid — APPROVED** [review]

**Comment by hhaAndroid** [comment]
lint failed

File: mmdet/models/dense_heads/rpn_head.py [hunk 9]
```diff
@@ -23,14 +24,28 @@ class RPNHead(AnchorHead):
...
     def _init_layers(self):
         """Initialize layers of the head."""
-        self.rpn_conv = nn.Conv2d(
-            self.in_channels, self.feat_channels, 3, padding=1)
+        if self.num_convs > 1:
+            rpn_convs = []
+            for i in range(self.num_convs):
+                rpn_convs.append(
+                    ConvModule(
+                        self.in_channels,
```
Comment (shinya7y) [inline mmdet/models/dense_heads/rpn_head.py:40]:
```python
in_channels = self.in_channels if i == 0 else self.feat_channels
rpn_convs.append(
    ConvModule(
        in_channels,
```

**Comment by levan92** [comment]
Hi, can I check why is this feature closed? Has it been merged into master?
